sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_serialize.py

```diff
@@ -170,15 +184,15 @@ def get_lease_action_properties(kwargs: Dict[str, Any]) -> Dict[str, Any]:
     }
 
 
-def get_cpk_info(scheme, kwargs):
-    # type: (str, Dict[str, Any]) -> CpkInfo
-    cpk = kwargs.pop('cpk', None)
+def get_cpk_info(scheme: str, kwargs: Dict[str, Any]) -> Optional[CpkInfo]:
+    cpk: Optional[CustomerProvidedEncryptionKey] = kwargs.pop('cpk', None)
     if cpk:
         if scheme.lower() != 'https':
             raise ValueError("Customer provided encryption key must be used over HTTPS.")
         return CpkInfo(
             encryption_key=cpk.key_value,
             encryption_key_sha256=cpk.key_hash,
-            encryption_algorithm=cpk.algorithm)
+            encryption_algorithm=cpk.algorithm  # type: ignore [arg-type]
```

Comment:
`CustomerProvidedEncryptionKey` inherits from `BlobCustomerProvidedEncryptionKey`, and the class variable `algorithm` technically has type `str`, but the expected type for the kwarg `encryption_algorithm` is `Literal['AES256']`.

What does everyone think about a temporary type ignore? I tried fixing this in blob, but the error won't go away.